samples/rust/rust_selftests.rs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Self test cases for Rust.
+
+use kernel::prelude::*;
+// Keep the "use" for a test in the test function. File-level uses are only for the test framework.
+
+module! {
+    type: RustSelftests,
+    name: b"rust_selftests",
+    author: b"Rust for Linux Contributors",
+    description: b"Self test cases for Rust",
+    license: b"GPL",
+}
+
+struct RustSelftests;
+
+/// A summary of testing.
+///
+/// A test can
+///
+/// * pass (successfully), or
+/// * hit an error (interrupted), sometimes it's a bug in the test case, or
+/// * fail (without hitting any error).
+///
+/// This is the type that differentiates the pass and fail case.
+#[allow(dead_code)]
+enum TestSummary {
```

Comment:
Elaborating a bit more on when to use `Ok(Fail)` vs. `Err` would be useful.

KUnit only has success/failure (and skipped).

Reply:
Fix the above three, and add more comment for `TestSummary` describing `Ok(Fail)` vs `Err`. Basically, `Err` means skipping because of hitting an error. And in the ci script, skipped tests are treated basically as failure. Sometimes skipping can be caused by a bug in the test case ;-)

Reply:
Much better, thank you! Just a tiny typo and we can merge this.